mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -467,10 +469,12 @@ object GeneralizedLinearRegression extends DefaultParamsReadable[GeneralizedLine
 
     override def variance(mu: Double): Double = mu * (1.0 - mu)
 
+    private def ylogy(y: Double, mu: Double): Double = {
+      if (y == 0) 0.0 else y * math.log(y / mu)
+    }
+
     override def deviance(y: Double, mu: Double, weight: Double): Double = {
```

Comment:
The utility method is not incorrect, I just think the above is more clear. That is how I've seen it done in e.g. statsmodels, and I think adding a static method for this simple logic is a bit superfluous. Isn't the entire point of the static method to avoid the y=0 case, which would calculate log(0), which is undefined?

Anyway, if we alternately elect to keep the `ylogy` method, I think a short doc would be helpful.